backend-api/src/functions/asyncActiveSession/tokenService/tokenService.ts

```diff
@@ -43,75 +59,74 @@ export class TokenService implements ITokenService {
       new Uint8Array(Buffer.from(encryptedCek, "base64")),
     );
     if (getCekResult.isError) {
-      return errorResult({
-        errorMessage: getCekResult.value.errorMessage,
-        errorCategory: getCekResult.value.errorCategory,
-      });
+      return getCekResult;
     }
 
     return successResult("");
   };
 
-  private readonly fetchPublicKey = async (
-    stsJwksEndpoint: string,
-  ): Promise<Result<JSON>> => {
-    let response;
-    try {
-      response = await fetch(stsJwksEndpoint, {
-        method: "GET",
-      });
-    } catch {
-      return errorResult({
-        errorMessage: "Unexpected error retrieving STS public key",
-        errorCategory: "SERVER_ERROR",
-      });
+  private readonly isJwks = (data: unknown): data is IJwks => {
+    return (
```

Comment:
Could this be simplified? 
```suggestion
  return (
      typeof data === "object" &&
      data !== null &&
      "keys" in data &&
      Array.isArray(data.keys) &&
      data.keys.every((key) => typeof key === "object" && key !== null)
    );
```

Reply:
I like `isJwks` 😄

Reply:
Thanks! updated